src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -581,6 +584,32 @@ public function isRelation($key)
                $this->relationResolver(static::class, $key);
     }
 
+    /**
+     * Determine if the given relation name is a BelongsTo relation.
+     *
+     * @param  string  $key
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function isBelongsToRelation($key, $value)
+    {
+        return $value instanceof Model
+            && $this->isRelation($key)
+            && $this->{$key}() instanceof BelongsTo;
+    }
+
+    /**
+     * Associates or attaches a model relation using the appropriate method relation.
+     *
+     * @param  string  $relation
+     * @param  \Illuminate\Database\Eloquent\Model  $instance
+     * @return $this
+     */
+    protected function fillBelongsToRelation($relation, $instance)
```

Comment:
I don't think this method is necessary. Are we expecting a user to override this?

The implementation is already very declarative, even if inlined on the call site.